include/rmm/mr/device/fixed_multisize_memory_resource.hpp

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <rmm/mr/device/device_memory_resource.hpp>
+#include <rmm/mr/device/fixed_size_memory_resource.hpp>
+#include <rmm/detail/error.hpp>
+#include <rmm/detail/aligned.hpp>
+
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/iterator/transform_iterator.h>
+
+#include <cuda_runtime_api.h>
+
+#include <list>
+#include <unordered_map>
+#include <cstddef>
+#include <utility>
+#include <algorithm>
+#include <cassert>
+#include <memory>
+
+#include <iostream>
+
+// forward decl
+using cudaStream_t = struct CUstream_st*;
+
+namespace rmm {
+
+namespace mr {
+
+/**
+ * @brief Allocates fixed-size memory blocks of a range of sizes.
+ * 
+ * Allocates blocks in the range `[min_size], max_size]` in power of two steps, 
```

Comment:
Since the sizes have to be powers of two anyways, considering doing something like CUB's allocator and have the min/max size specified as an exponent. Likewise, could customize the growth factor to be something other than 2. 

See https://github.com/rapidsai/rmm/blob/7957b39dc403b3b0965fef37f6b867c1b68c577c/include/rmm/mr/device/cub_memory_resource.hpp#L51

Reply:
Done.